compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/FuseAndHoistParallelLoops.cpp

```diff
@@ -36,36 +43,58 @@ struct FuseAndHoistParallelLoopsPass final
 };
 } // namespace
 
-struct FuseForalls final : OpRewritePattern<tensor::ExtractSliceOp> {
+static bool forallTripCountMatchesWorkgroupSize(scf::ForallOp forallOp,
+                                                int64_t flatWorkgroupSize) {
+  // True for lane mapped loops.
+  if (forallOpHasMappingType<IREE::GPU::LaneIdAttr>(forallOp)) {
+    return true;
+  }
```

Comment:
Why does this automatically return true for lane mapped loops? Shouldn't it check that it either matches the workgroup size or has a parent warp forall with which it has a product that matches the workgroup size?